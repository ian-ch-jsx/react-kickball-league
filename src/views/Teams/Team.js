import { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import { getTeamById } from '../../services/teams';
import './Teams.css';

export default function Team(props) {
  const [team, setTeam] = useState([]);
  const [loading, setLoading] = useState(true);
  useEffect(() => {
    const fetchData = async () => {
      const data = await getTeamById(props.match.params.id);
      setTeam(data[0]);
      setLoading(false);
    };
    fetchData();
  }, [props.match.params.id]);
  if (loading) return <h1>loading</h1>;

  return (
    <div className="container">
      <h1>{team.name}</h1>
      <h2>Players</h2>
      {team.players.map((player) => (
        <p key={player.id}>
          <Link to={`/players/${player.id}`} key={player.id}>
            {player.name}
          </Link>
        </p>
      ))}
    </div>
  );
}
