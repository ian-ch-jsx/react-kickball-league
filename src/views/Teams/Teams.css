.container {
  display: flex;
  flex-direction: column;
  line-height: 20px;
}

h1 {
  margin: 20px;
}

h2 {
  margin: 10px;
}
