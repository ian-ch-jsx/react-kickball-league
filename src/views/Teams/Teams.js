// Fetch team Data
// map through teams to display link to team page
import { useEffect, useState } from 'react';
import { getTeams } from '../../services/teams';

export default function Teams() {
  const [teams, setTeams] = useState([]);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    const fetchData = async () => {
      const data = await getTeams();
      setTeams(data);
      setLoading(false);
    };
    fetchData();
  }, []);

  if (loading) return <h1>loading</h1>;
  return (
    <ul>
      {teams.map((team) => (
        <li key={team.id}>
          <p>{team.name}</p>
        </li>
      ))}
    </ul>
  );
}
