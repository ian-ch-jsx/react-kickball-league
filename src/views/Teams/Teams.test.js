import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Teams from './Teams';

test('renders list of teams', async () => {
  const { container } = render(
    <MemoryRouter>
      <Teams />
    </MemoryRouter>
  );

  await screen.findByText('Bridge City Sneakers', 'VanCougers');

  expect(container).toMatchSnapshot();
});
