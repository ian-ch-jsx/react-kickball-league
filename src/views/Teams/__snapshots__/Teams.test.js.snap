// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders list of teams 1`] = `
<div>
  <div
    class="container"
  >
    <h1>
      Teams
    </h1>
    <p>
      <a
        href="/teams/1"
      >
        Bridge City Sneakers
      </a>
    </p>
    <p>
      <a
        href="/teams/3"
      >
        Lakeville Thunderfeet
      </a>
    </p>
    <p>
      <a
        href="/teams/4"
      >
        Mt. Hoodies
      </a>
    </p>
    <p>
      <a
        href="/teams/2"
      >
        Stumptown Lumberjacks
      </a>
    </p>
    <p>
      <a
        href="/teams/5"
      >
        Vancougers
      </a>
    </p>
  </div>
</div>
`;
